lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py

```diff
@@ -46,8 +59,35 @@ def main(event, _):
     print(f'event = {event!r}')
 
     ec2_client = boto3.client('ec2')
+    s3_client = boto3.client('s3')
+
+    bucket_name = os.environ["BUCKET_NAME"]
+    object_path = os.environ["OBJECT_PATH"]
 
     ec2_instance_id = event['detail']['ec2InstanceId']
 
-    response = create_tags(ec2_client, ec2_instance_id, event['detail'])
-    print(f'response = {response!r}')
+    try:
+        s3_client.get_object(
```

Comment:
We could use `head_object()` here to be slightly more efficient – we’re only interested in checking whether it exists, not what the file contains.

```pycon
>>> import boto3
>>> s3 = boto3.client('s3')
>>> s3.head_object(Bucket='platform-infra', Key='tmp/doesnotexist')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/chana/.virtualenvs/api/lib/python3.6/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/chana/.virtualenvs/api/lib/python3.6/site-packages/botocore/client.py", line 599, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (404) when calling the HeadObject operation: Not Found
```

Although note you’ll need to tweak the exception handling – specifically, I think you’d want

```python
if ex.response['Error']['Code'] == '404':
    ...
```